server/agents/ll_word_suggest_upgrade_agent.py

```diff
@@ -0,0 +1,223 @@
+
+Recently Suggested: `{live_upgrade_word_history}`
+
+Output Format: {format_instructions}
+
+Don't output punctuation or periods! Output all lowercase! Define 1/5 of the words in the input text (never define all of the words in the input, never define highly common words like "the", "a", "it", etc.). Now provide the output:
+"""
+
+#opposite language (either {source_language} or {target_language}, whatever is
+
+#This level influences the selection of upgrade word:
+#   - 0-49 (Beginner): suggest Upgrade word that is less advanced, meaning percentile rank >0.15 (people use in daily life but are not used by learner).
+#   - 50-74 (Conversational): suggest more advanced Upgrade, meaning percentile rank >1.5 (people do not use in Input Language, exclusively exists in Output Language in a unique way, like idiom/ culturally relevant phrases).
+#   - 75-100 (Intermediate): only suggest Upgrade that is very rare words, meaning percentile rank >30, which appears in studies/ researches/ academic contexts/ ancient language that neither Input language user nor Output Language user use in ordinary lives.
+
+#A word with frequency percentile 1.5 is a little uncommon, a word with percentile 0.15 is very common, a word with percentile >=30 is super rare. Use the percentile to determine words that the user might not know, where higer number is more rare.
+
+#*** Never do intralanguage translation (don't translate within the same language).
+# using the format instructions above:
+#
+#There should be 1 words per run in the dict. Don't output any explanation or extra data, just this simple info. It's OK to output zero words if there are no appopriately rare words in the input text.
+
+#Don't suggest stop words or super common words like "yes", "no", "he", "hers", "to", "from", "thank you", "please", etc. in ANY language, but you can define even semi-common words like "exactly", "bridge", "computer", etc. for beginners with a fluency level less than 50 (current user fluency is {fluency_level}.
+
+#Don't pick words already in the conversation! Never suggest words in Input Text! Give an upgrade word (more advanced and related synonym, idiom, or rare word)
+
+#Conversation 4: "that museum’s architecture is very beautiful."
+#Source Language 4: English
+#Target Language 4: Chinese (Pinyin)
+#Output 4: {{ "upgrade": "gǔdài","meaning": "ancient"}}
+
+
+#Preface Rule:
+#Use Pinyin when writing Chinese. Never use Chinese characters, use Pinyin.
+#, use Pinyin if writing Chinese)
+#3.a. If writing Chinese, output exclusively in Pinyin, avoiding Chinese characters entirely.
+#NEVER OUTPUT CHINESE CHARACTERS. USE PINYIN, USE THE LATIN ALPHABET PINYIN FOR CHINESE.
+
+#Use Pinyin when writing Chinese. Never use Chinese characters, use Pinyin.
+
+# in_upgrade_translation must be in Pinyin or Latin characters.
+
+# If writing Chinese, output exclusively in Pinyin, avoiding Chinese characters entirely.
```

Comment:
There's a lot of extra comment that I assume come from the original ll file, you can remove them from here they aren't needed